cityofaustin/atd-moped
# Remove component edit mode dialog

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/13243

This PR removes the `EditModeDialog` that displayed after choosing to edit a component in the Map interface. I remapped a lot of the state handling to happen one step before it used to. The clicking of a component list item nows takes on setting both `clickedComponent` state (to track which features to highlight and which item to expand) and `draftEditComponent` (to capture the edits being made to the currently selected component).

## Testing
**URL to test:** 

https://deploy-preview-1109--atd-moped-main.netlify.app/moped/

**Steps to test:**
1. Create a project with the **Use signal asset** toggle and selecting a signal. Save and then go to the **Map** tab.
2. Move the map away from the signal component and then click the magnifier icon and the map should move back to the signal feature
3. Click the signal component list item on the left side of the screen and the list item should expand to show **Delete** and **Attributes** buttons.
4. Click **Attributes** and update different combinations of fields and save each time. The form modal should disappear when **Save** is clicked, and the list item for the signal component should remain expanded.
5. Click the map when a list item is expanded and the list item should collapse (the draft component held in state is also cleared behind the scenes).
6. Click the top of a list items and it should expand and close each time you click (the initial click to expand a list item loads the component details into state for attribute or features edits).
7. Add a component with type **Bike Lane - Climbing** and map it
8. Click the new component list item, and you should see it expand and show **Delete**, **Attributes**, and **Map** buttons.
9. Click **Attributes** and try editing different combinations of fields.
10. Click **Map** and update the component's features
11. Click **Delete** to remove the component from the project.
12. Repeat steps 7 to 11 with a point type component like **Bike Box**
13. Also try clicking around different component list items to make sure they highlight in blue and the other components are greyed out

---
#### Ship list
